lib/engine/game/g_18_india/game.rb

```diff
@@ -761,25 +802,25 @@ def commodity_bonus(route, _stops)
               revenue += bonus
             when 'SPICES'
               # SPICES => KOCHI [G36] + 70
+              if visited_names.include?('KOCHI')
+                bonus = 70
               # SPICES => COLOMBO [K40] + 50
               # SPICES => CHENNAI [K30] + 50
+              elsif visited_names.include?('COLOMBO') ||
+                    visited_names.include?('CHENNAI')
```

Comment:
This all could be clearer if you declare a hash for the various spice bonuses. Something like:
```
SPICE_BONUSES = {
    'KOCHI' => 70,
    'COLOMBO' => 50,
    'CHENNAI' => 50,
    'LAHORE' => 40,
    # …
}.freeze
```
And then, if you're looking for the maximum value visited location, you'd have something like this:
`visited_names.map { |loc| SPICE_BONUSES[loc] || 0 }.max`
[I feel that there should be a more elegant way of writing this].

Reply:
I like your ideas here.  Intersect looks cleaner. 

I will need to refactor some of this code later, as the Jewelry Concession can add a wrinkle to my current implementation.
I will verify with the designers that Jewelry could be on the same hex as other commodities, in which case I think it may be difficult for hex//tile to have more than one name.   Once I get the graphics from them, I'll add icons to the hexes, which I can use for the comparison, or simply use the hex ID.   I did like using the words / names as the code was easier to read and verify.